users_ldap_populate/models/users_ldap.py

```diff
@@ -69,13 +69,29 @@ def action_populate(self):
                     conf['ldap_filter'])
             results = self.get_ldap_entry_dicts(conf)
             for result in results:
+                login = result[1][login_attr][0].lower().strip()
                 user_id = self.with_context(
                     no_reset_password=True
-                ).get_or_create_user(conf, result[1][login_attr][0], result)
-                # this happens if something goes wrong while creating the user
-                # or fetching information from ldap
+                ).get_or_create_user(conf, login, result)
                 if not user_id:
-                    deactivate_unknown = False
+                    # this happens if the user exists but is active = False
+                    # -> fetch the user again and reactivate it
+                    self.env.cr.execute(
+                        "SELECT id, active FROM res_users "
+                        "WHERE lower(login)=%s",
+                        (login,))
+                    res = self.env.cr.fetchone()
+                    if res:
+                        user_id, active = res
+                        self.env['res.users'].sudo().browse(
```

Comment:
indeed. I'd also suggest to call `get_or_create_user` again in this case, to update the user with the newest values from ldap. And thanks, this was my mistake from v8 I think :-)

Reply:
@hbrunn get_or_create_user does not update the user if it exists

Reply:
@simahawk fixed, good catch!

Reply:
ah, that's just something I do: https://github.com/OCA/server-tools/blob/10.0/users_ldap_groups/models/res_company_ldap.py

Ignore then, as this is called periodically it will be called anyways some time later. It might be worthwhile to write the result of `map_ldap_values` on the user for cases where the user came from another company or has other values changed in ldap. Maybe we should do this anyways for all users during populate? (that's more a general question, not a feature request on your PR)

Reply:
indeed. Totally out of scope for this PR, but seems like a useful extension of this module to me